packages/integration-karma/test/misc/clean-dom/index.spec.js

```diff
@@ -0,0 +1,21 @@
+describe('clean DOM', () => {
+    it('inserts elements to dirty the DOM', () => {
+        // insert to the body
+        const el = document.createElement('div');
+        el.className = 'yolo';
+        document.body.appendChild(el);
+        expect(document.querySelector('.yolo')).toEqual(el);
+
+        // insert to the head as well
+        const style = document.createElement('style');
+        style.textContent = 'html { background-color: green }';
+        style.id = 'the-style';
+        document.head.appendChild(style);
+        expect(document.querySelector('#the-style')).toEqual(style);
+    });
+
+    it('DOM is clean after each test', () => {
+        expect(document.querySelector('.yolo')).toBeNull();
```

Comment:
I wanted to know if this could lead to false-positives because it is dependent on the order of execution. So I tried to find out if Jasmine executes tests in the same spec in order but didn't find much info.
[The doc](https://jasmine.github.io/api/3.7/Configuration.html) had a `random` flag which determines `Whether to randomize spec execution order`. This might be randomizing the spec itself and not the test cases within a spec. 🤷‍♂️

Reply:
You're right; the order is unpredictable. When I refresh the page, it seems to randomly make one test run before the other. I'll see if I can make it consistent.

Reply:
I added `random: false` to the Jasmine config and it seems to work. Test execution is based on the order you declare them in the file. I refreshed the page like 10 times and tried switching them around, and it seemed to always do the same ordering. 🙂